pkg/sql/schemaexpr/partial_index.go

```diff
@@ -214,12 +214,13 @@ func FormatIndexForDisplay(
 	}
 
 	if index.IsPartial() {
-		f.WriteString(" WHERE ")
 		pred, err := FormatExprForDisplay(ctx, table, index.Predicate, semaCtx, tree.FmtParsable)
 		if err != nil {
 			return "", err
 		}
+		f.WriteString(" WHERE (")
 		f.WriteString(pred)
+		f.WriteString(")")
```

Comment:
Let's remove these so that `SHOW CREATE` looks like PG's `\d table`:

```
marcus=# \d t
                 Table "public.t"
 Column |  Type   | Collation | Nullable | Default
--------+---------+-----------+----------+---------
 k      | integer |           | not null |
 a      | integer |           |          |
Indexes:
    "t_pkey" PRIMARY KEY, btree (k)
    "i" btree (k) WHERE a > 0
```

Reply:
sounds good to me